sdcm/nemesis.py

```diff
@@ -1074,6 +1074,35 @@ def disrupt_modify_table(self):
         disrupt_func = getattr(self, disrupt_func_name)
         disrupt_func()
 
+    def disrupt_mgmt_backup(self):
+        self._set_current_disruption('ManagementBackup')
+        if not self.cluster.params.get('use_mgmt', default=None):
+            raise UnsupportedNemesis('Scylla-manager configuration is not defined!')
+
+        manager_node = self.monitoring_set.nodes[0]
+        manager_tool = mgmt.get_scylla_manager_tool(manager_node=manager_node)
+
+        cluster_name = self.cluster.name
+        mgr_cluster = manager_tool.get_cluster(cluster_name)
+        if not mgr_cluster:
+            self.log.debug("Could not find cluster : {} on Manager. Adding it to Manager".format(cluster_name))
+            ip_addr_attr = 'public_ip_address' if self.cluster.params.get('cluster_backend') != 'gce' and \
+                Setup.INTRA_NODE_COMM_PUBLIC else 'private_ip_address'
+            targets = [getattr(n, ip_addr_attr) for n in self.cluster.nodes]
+            mgr_cluster = manager_tool.add_cluster(name=cluster_name, host=targets[0], disable_automatic_repair=True,
+                                                   auth_token=self.monitoring_set.mgmt_auth_token)
+        temp_bucket_location = 'manager-backup-tests-eu-west-1'
+        for node in self.cluster.nodes:
+            update_config_file(node=node, region='eu-west-1')
+        mgr_task = mgr_cluster.create_backup_task({'location': ['s3:{}'.format(temp_bucket_location)]})
+        mgr_task.wait_for_status([TaskStatus.DONE], timeout=86400)  # timeout is 24 hours
```

Comment:
Timeout is too big, it’s should be either decided dynamically based on data set size or better probe the task progress in a loop and make sure it’s progressing.

Reply:
decreased it to 12 hours, as the progress could take very long time in some cases, or either fail from time to time (and it will resume, as it has an internal retry mechanism)